girder/models/folder.py

```diff
@@ -79,9 +79,11 @@ def validate(self, doc, allowRename=False):
                                   doc['parentCollection'],
                                   'girder.models.folder.invalid-parent-type')
         name = doc['name']
+        # If the folder already exists with the current name, don't check.
+        checkName = '_id' not in doc or not self.findOne({'_id': doc['_id'], 'name': name})
```

Comment:
I'm assuming the query engine is smart enough that this will not be asymptotically slower than a normal `_id` query? If it is for some reason slower, we should just do the findOne and then check the name in python.

Reply:
Both `name` and `_id` have indexes, and there's no strange compound logic here. I'd be very surprised if this didn't scale properly.

Reply:
I would also be very surprised; just the presence of a unique index field (_id) in a findOne query should mean it's nearly identical in query time to an _id alone. Just wanting to sanity check.